### PR TITLE
PIM-4079 : mongodb completeness

### PR DESCRIPTION
**Pull request by jmleroux**
| Q                    | A
| -------------------- | ---
| Bug fix?             |n
| New feature?         |n
| BC breaks?           |n
| CI currently passes? |n
| Tests pass?          |y
| Scenarios pass?      |y
| Checkstyle issues?*  |n
| PMD issues?**        |n
| Changelog updated?   |n
| Fixed tickets        |PIM-4079
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php [hunk 1]
```diff
@@ -183,6 +170,46 @@ function ($value) {
...
+        $attributesReqs = $normalizedReqs[$missingComp]['reqs']['attributes'];
+        $pricesReqs     = $normalizedReqs[$missingComp]['reqs']['prices'];
+
+        return count($attributesReqs) + count($pricesReqs);
+    }
+
+    /**
+     * @param array  $normalizedReqs
+     * @param string $missingComp
+     * @param        $dataFields
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: seems to be an array

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php [hunk 2]
```diff
@@ -183,6 +170,46 @@ function ($value) {
...
+     *
+     * @return int
+     */
+    protected function getMissingCount($normalizedReqs, $missingComp, $normalizedData, $dataFields)
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: typehint normalizedReqs and dataFields as array

**Comment by nidup** [comment]
GTM

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php (hunk 2 above)
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: one arg missing in the phpdoc

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php [hunk 3]
```diff
@@ -183,6 +170,46 @@ function ($value) {
     }
 
     /**
+     * @param array  $normalizedReqs
+     * @param string $missingComp
+     *
+     * @return int
+     */
+    protected function getRequiredCount($normalizedReqs, $missingComp)
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/CompletenessGenerator.php]: typehint

**Pull request merged**